core/src/main/java/io/questdb/griffin/engine/functions/str/LPadStrFunctionFactory.java

```diff
@@ -0,0 +1,127 @@
+import io.questdb.std.ObjList;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.Nullable;
+
+public class LPadStrFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "lpad(SIS)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        final Function strFunc = args.getQuick(0);
+        final Function lenFunc = args.getQuick(1);
+        final Function fillTextFunc = args.getQuick(2);
+        return new LPadStrFunc(strFunc, lenFunc, fillTextFunc);
+    }
+
+    public static class LPadStrFunc extends StrFunction implements TernaryFunction {
+
+        private final Function strFunc;
+        private final Function lenFunc;
+        private final Function fillTextFunc;
+
+        private final StringSink sink = new StringSink();
+        private final StringSink sinkB = new StringSink();
+
+        public LPadStrFunc(Function strFunc, Function lenFunc, Function fillTexFunc) {
+            this.strFunc = strFunc;
+            this.lenFunc = lenFunc;
+            this.fillTextFunc = fillTexFunc;
+        }
+
+        @Override
+        public Function getLeft() {
+            return strFunc;
+        }
+
+        @Override
+        public Function getCenter() {
+            return lenFunc;
+        }
+
+        @Override
+        public Function getRight() {
+            return fillTextFunc;
+        }
+
+        @Override
+        public CharSequence getStr(final Record rec) {
+            return lPadStr(strFunc.getStr(rec), lenFunc.getInt(rec), fillTextFunc.getStr(rec), sink);
+        }
+
+        @Override
+        public CharSequence getStrB(final Record rec) {
+            return lPadStr(strFunc.getStr(rec), lenFunc.getInt(rec), fillTextFunc.getStr(rec), sinkB);
+        }
+
+        @Override
+        public int getStrLen(Record rec) {
+            final CharSequence str = strFunc.getStr(rec);
+            final int len = lenFunc.getInt(rec);
+            final CharSequence fillText = fillTextFunc.getStr(rec);
+            if (str != null && len != Numbers.INT_NaN && len >= 0 && fillText != null && fillText.length() > 0) {
+                return len;
+            } else {
+                return TableUtils.NULL_LEN;
+            }
+        }
+
+        @Nullable
+        private static StringSink lPadStr(CharSequence str, int len, CharSequence fillText, StringSink sink) {
+            if (str != null && len != Numbers.INT_NaN && len >= 0 && fillText != null && fillText.length() > 0) {
+                sink.clear();
+                if (len > str.length()) {
+                    final int fillTextLen = fillText.length();
+                    for (int i = 0; i < (len - str.length()); i++) {
+                        sink.put(fillText.charAt(i % fillTextLen));
```

Comment:
Modulo is an expensive operation and it's better to avoid calling it in the loop body. I'd better replace it with something like the following:
```java
        for (int i = 0, n = (len - str.length()) / fillTextLen; i < n; i++) {
            sink.put(fillText);
        }
        for (int i = 0, n = (len - str.length()) % fillTextLen; i < n; i++) {
            sink.put(fillText.charAt(i));
        }
```